rest-server/core/src/main/scala/com/actionml/core/template/Engine.scala

```diff
@@ -50,7 +50,7 @@ abstract class Engine extends LazyLogging with JsonParser {
     }
   }
 
-  def init(json: String): Validated[ValidateError, Boolean] = parseAndValidate[RequiredEngineParams](json)
+  def init(json: String): Validated[ValidateError, Boolean] = parseAndValidate[GenericEngineParams](json)
```

Comment:
I don't like vars (engineId and mirroring) and 'init' methods, and I think we can get rid of this later with something like 'factory method' pattern:
```Engine { def engineId }
CBEngine(eid: String) extends Engine { override def engineId = eid }
object CBEngine.apply(json: String)
```
but i'm just mentioning todos for future

Reply:
We use it so Validated can be returned from Init since this is the only place the JSON Engine params file is validated and parsed. Apply would not allow for this. But we can discuss.